### DISCUSSION TITLE
Is it possible to configure a Pydantic Model to leave field values "as is", if conversion caused a ValidationError?

### DISCUSSION THREAD

**Discussion by SaumyaKhati** [pydantic/pydantic#4284] [Question] [answered]
Is it possible to configure pydantic models so that they ignore field values that resulted in a Validation error?

Ex: if we have...
```python
from pydantic import BaseModel

class User(BaseModel):
    a: int
    b: list

test_event = {
    "a": "4",
    "b": 4.20,
}

User(**test_event).dict()  # This should return {"a": 4, "b": 4.20 }. 
```

Essentially, I want the model to leave field values unchanged if it would cause a validation error, otherwise convert it as normal.

I've tried various ad-hoc solutions of using root validators, custom validators (w/ pre=True) but still I've had no success.
I've tried searching online through various StackOverFlow posts but still haven't found anything regarding this use case.

**Comment by SaumyaKhati** [comment]
One potential way I could think of doing this is by defining a custom validator (for **all** fields) that tries to do the casting, and if failed, just returns the value "as is". But I keep running into the issue of **other** (standard) validators "failing". Is there a way to restrict the pydantic model so that **only my custom validator runs** for the model's fields? Using pre=True/False did not seem to work (both for the field validator or the root validator). 

When trying to work with the root_validator, I tried to clear the \_\_validators\_\_ field of the model, but this did not work either.

**Comment by samuelcolvin** [comment, marked as answer]
Please format example to make your query easy to read.

**Reply by SaumyaKhati** [reply]
Done! Could you please take a look again Samuel? Thanks!

**Reply by samuelcolvin** [reply]
No, No, No that's much worse. Please take 10 minutes to learn about [github markdown](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks).

What you're asking for is not really possible in pydantic V1, It'll be possible with V2.

**Reply by samuelcolvin** [reply]
Yes you could use a custom validator, but you'd loose a lot of the power of pydantic.

**Reply by SaumyaKhati** [reply]
My apologies! I've fixed the example. Ok, if it's not possible as a "built-in" option w/ V1, could you please explain how this would be possible w/ a custom validator (despite the stated consequences)? 

In the answer above, I've explained my attempt to use custom validators but it did not seem to work because the other "standard" validators still run (and fail). Is it possible to restrict the model to **only** run my custom validator?

**Reply by samuelcolvin** [reply]
Well that's the point, in V2 you can use [wrap validators](https://pydantic-docs.helpmanual.io/blog/pydantic-v2/#validator-function-improvements) which would allow you to either skip validation or catch the error from validation and use another value.

In V1, the only solution would be to use a [custom data type](https://pydantic-docs.helpmanual.io/usage/types/#custom-data-types) and call a validator imported from [`validators.py`](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/validators.py), then catch the error. But it would be a lot of work.

**Reply by SaumyaKhati** [reply]
Ah, I see. Ok, that makes sense! Thank you very much.

**Comment by SaumyaKhati** [comment]
@samuelcolvin How can I define custom types that override typing.Dict and typing.List? I tried to define a data type (ex. "MyList") that inherit from the typing.List class directly but it was not allowed due to "not being a generic class error", and then I tried to define a class (call it "MyDict") that inherited from dict and MutableMapping[KV, VT] (which is what the official docs shows for typing.Dict), and while this did not cause any errors, I could not utilize my own "validate" method because it was not being called for my datatype. But the values were still being validated? 

What is the validation procedure for more complex types like Dict/List and how can I create custom types that would catch validation for these types? If you have an idea, would it be possible to see a working example please? It would be greatly appreciated.
